src/libstore/builtins/unpack-channel.cc

```diff
@@ -0,0 +1,39 @@
+#include "rust.hh"
+#include "builtins.hh"
+#include "compression.hh"
+
+namespace nix {
+
+void builtinUnpackChannel(const BasicDerivation & drv)
+{
+    auto getAttr = [&](const string & name) {
+        auto i = drv.env.find(name);
+        if (i == drv.env.end()) throw Error("attribute '%s' missing", name);
+        return i->second;
+    };
+
+    Path out = getAttr("out");
+    auto channelName = getAttr("channelName");
+    auto src = getAttr("src");
+
+    createDirs(out);
+
+    auto source = sinkToSource([&](Sink & sink) {
+        auto decompressor =
+            hasSuffix(src, ".bz2") ? makeDecompressionSink("bzip2", sink) :
+            hasSuffix(src, ".xz") ? makeDecompressionSink("xz", sink) :
```

Comment:
No gzip?

Reply:
Yeah, we currently lack gzip support in `compression.cc`. Probably not a big deal since our channels don't use it, but it would be nice to add.